prov/efa/src/efa_mr.c

```diff
@@ -845,6 +845,7 @@ static uint64_t efa_mr_cuda_non_p2p_keygen(void) {
 static int efa_mr_reg_impl(struct efa_mr *efa_mr, uint64_t flags, void *attr)
 {
 	uint64_t core_access, original_access;
+	void *hmem_data = NULL;
```

Comment:
i still think we can simply this by passing `attr` instead of a pointer. This function is an internal function called by `efa_mr_cache_entry_reg` and `efa_mr_regattr`, so we should be good to do it.

Reply:
I'm not sure about that. Passing the struct will cause an implicit copy right?

Reply:
are other variables in `smr_env` configurable?